app/src/main/java/com/example/tasky/agenda_details/data/ImageCompressionImpl.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.tasky.agenda_details.data
+
+import android.content.ContentResolver
+import android.graphics.Bitmap
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import com.example.tasky.agenda_details.domain.ImageCompressionUseCase
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.ByteArrayOutputStream
+
+class ImageCompressionUseCaseImpl(private val contentResolver: ContentResolver) :
+    ImageCompressionUseCase {
```

Comment:
While the way you did this is totally correct, a use case is meant for business logic and should be in domain. So instead you'd just need to give this a different name, like `ImageCompressor` and `BitmapCompressor` for the implementation 👍

Reply:
ah true!